elasticsearch_dsl/_async/search.py

```diff
@@ -16,28 +16,32 @@
 #  under the License.
 
 import contextlib
+from typing import AsyncIterator, Generic
 
 from elasticsearch.exceptions import ApiError
 from elasticsearch.helpers import async_scan
+from typing_extensions import TypeVar
 
 from ..async_connections import get_connection
-from ..response import Response
+from ..response import Hit, Response
 from ..search_base import MultiSearchBase, SearchBase
 from ..utils import AttrDict
 
+_R = TypeVar("_R", default=Hit)
```

Comment:
This `_R` typevar is inserted into pretty much every search or response related class. I've used the version of `TypeVar` from typing_extensions because it allows you to use a default.